src/code/z_player_lib.c

```diff
@@ -1076,132 +1148,93 @@ void Player_DrawGetItem(GlobalContext* globalCtx, Player* this) {
     }
 }
 
-void func_80090A28(Player* this, Vec3f* vecs) {
-    D_8012608C.x = D_80126080.x;
+void Player_GetSwordTip(Player* this, Vec3f* swordTip) {
+    sSwordTipPos2.x = sSwordTipPos1.x;
```

Comment:
this is also used in the context of sticks as well. I am 100% ok with grouping sticks in with swords, cause it is basically a reskinned sword. Just wanted to make sure we are all on board with that

Reply:
I'm very not familiar with the player code but doesn't it also get used with hammer (and maybe fishing rod?)

Reply:
it will be used by anything that sets sword state to non zero.
so fishing rod no, hammer yes

if we want to address that term and make it more broad, `swordState` and related vars will need to be addressed as well lol

Reply:
but as far as players internal systems are concerned, hammer is just a shorter stubbier sword with slightly different properties. so i personally dont have an issue with it lol. but can understand if others do

Reply:
what about `meleeWeapon` (I was looking for a synonym of "close combat weapon" and wikipedia saved the day https://en.wikipedia.org/wiki/Melee_weapon )

Reply:
not a fan of the vagueness that comes from that name, sword is fine imo

Reply:
I would say this falls in line with the "player boots," which are not actually always boots (link has different "boots" if he's inside so he walks slower), but I was still told to call it boots, so sword here should be fine too I think since that's the main purpose of it, but it gets reused for other things

Reply:
Well yeah the point is to be more vague so it is also accurate for non-swords melee weapons like stick and hammer

But it is also precise enough (and actually not vague) because swords, stick and hammer are all the melee weapon available to Link.

Reply:
I or someone else will open a separate PR for sword -> MeleeWeapon (after more discussion eventually) since such a change (that you aren't eager to do anyway 😅) would require touching more than this PR touches, and this PR is big enough already.